unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Passport/Passport/PassportNavigation/ViewAllComponentController.cs

```diff
@@ -0,0 +1,276 @@
+    {
+        view.ShowNftIcons(iconsWithWearables);
+    }
+
+    public void Dispose()
+    {
+        view.OnBackFromViewAll -= BackFromViewAll;
+        view.OnRequestCollectibleElements -= RequestCollectibleElements;
+
+        sectionsCts.SafeCancelAndDispose();
+        sectionsCts = null;
+    }
+
+    private void RequestCollectibleElements(PassportSection section, int pageNumber, int pageSize)
+    {
+        view.CloseAllNftItemInfos();
+
+        switch (section)
+        {
+            case PassportSection.Wearables:
+                RequestOwnedWearablesAsync(currentPlayerId, pageNumber, pageSize, sectionsCts.Token).Forget();
+                break;
+            case PassportSection.Names:
+                RequestOwnedNamesAsync(currentPlayerId, pageNumber, pageSize, sectionsCts.Token).Forget();
+                break;
+            case PassportSection.Lands:
+                RequestOwnedLandsAsync(currentPlayerId, pageNumber, pageSize, sectionsCts.Token).Forget();
+                break;
+        }
+    }
+
+    private async UniTask RequestOwnedWearablesAsync(string userId, int pageNumber, int pageSize, CancellationToken ct)
+    {
+        try
+        {
+            view.SetLoadingActive(true);
+
+            (IReadOnlyList<WearableItem> wearables, int totalAmount) ownedWearableItems =
+                await wearablesCatalogService.RequestOwnedWearablesAsync(userId, pageNumber, pageSize, cleanCachedWearablesPages, ct);
+
+            cleanCachedWearablesPages = false;
+            ProcessReceivedWearables(ownedWearableItems.wearables);
+            view.SetTotalElements(ownedWearableItems.totalAmount);
+            view.SetLoadingActive(false);
+        }
+        catch (Exception e) when (e is not OperationCanceledException)
+        {
+            ShowErrorAndGoBack();
+        }
+    }
+
+    private async UniTask RequestOwnedNamesAsync(string userId, int pageNumber, int pageSize, CancellationToken ct)
+    {
+        try
+        {
+            view.SetLoadingActive(true);
+
+            using var pagePointer = namesService.GetPaginationPointer(userId, pageSize, CancellationToken.None);
+            var response = await pagePointer.GetPageAsync(pageNumber, ct);
+
+            if (response.success)
+            {
+                ProcessReceivedNames(response.response.Names);
+                view.SetTotalElements(response.response.TotalAmount);
+                view.SetLoadingActive(false);
+            }
+            else
+                ShowErrorAndGoBack();
+        }
+        catch (Exception e) when (e is not OperationCanceledException)
+        {
+            ShowErrorAndGoBack();
+        }
+    }
+
+    private async UniTask RequestOwnedLandsAsync(string userId, int pageNumber, int pageSize, CancellationToken ct)
+    {
+        try
+        {
+            view.SetLoadingActive(true);
+
+            using var pagePointer = landsService.GetPaginationPointer(userId, pageSize, CancellationToken.None);
```

Comment:
I want to emphasize how it works:
every time you call `GetPaginationPointer` it returns a new `cache`: I want to ensure it is desirable behavior: because thus you don't cache any results at all. So when you request the page you requested before it will fetch it again from the server.

Reply:
In my opinion you should do it similarly to `cleanCachedWearablesPages`

Reply:
Totally agree... in the same way I did for the `Wearables` service, I should manage a cache also for `Names` and `Lands` services.

I have taken advantage of this necessity to apply a refactor of both services and make them more similar to the `Wearables` one, where we can perform the data request through a single method (`RequestOwnedNamesAsync` for names and `RequestOwnedLandsAsync` for lands) instead of 2:

### BEFORE
```
using var pagePointer = namesService.GetPaginationPointer(<userId>, <pageSize>, ct);
names = await pagePointer.GetPageAsync(<pageNum>, ct).Elements;
```

```
using var pagePointer = landsService.GetPaginationPointer(<userId>, <pageSize>, ct);
lands = await pagePointer.GetPageAsync(<pageNum>, ct).Elements;
```

### NOW
```
var names = await namesService.RequestOwnedNamesAsync(<userId>, <pageNum>, <pageSize>, <cleanCachedWearablesPages>, ct);
```

```
var lands = await landsService.RequestOwnedLandsAsync(<userId>, <pageNum>, <pageSize>, <cleanCachedWearablesPages>, ct);
```

You can take a look to the change here: https://github.com/decentraland/unity-renderer/pull/4316/commits/313e3620d87b9a44e89c692c77b7088055b79789